checker.cpp

```diff
@@ -1,17 +1,49 @@
 #include <assert.h>
+#include <iostream>
 
-bool vitalsAreOk(float bpm, float spo2, float respRate) {
-  if(bpm < 70 || bpm > 150) {
-    return false;
-  } else if(spo2 < 90) {
+
+void raiseSoundAlert (const char * vitalName, const char* level)
+{
+  std::cout << "Sound: " << vitalName << " is " << level << std::endl;
+}
+
+bool level_is_ok (const char* name, float value, float lower_limit, float upper_limit)
+{
+
+  if (value<lower_limit)
+  {
+   raiseSoundAlert (name, "LOW");
```

Comment:
Try polymorphic behavior either with function pointer or virtual function, instead of directly coupling the implementation to raiseSoundAlert.